### PR TITLE
Optimize `number_small` capture

### PR DESCRIPTION
**Pull request by pokey**
This optimisation reduces DFA from >1100ms to <500ms when combined with Cursorless main branch

All credit due to @AndreasArvidsson, I made only tiny tweaks to resolve conflicts with knausj

### CODE REVIEW

File: core/numbers/numbers.py [hunk 1]
```diff
@@ -5,12 +5,12 @@
 mod = Module()
 ctx = Context()
 
-digits = "zero one two three four five six seven eight nine".split()
+digit_list = "zero one two three four five six seven eight nine".split()
```
Comment (pokey) [inline core/numbers/numbers.py:8]: This variable name conflicts with the `digits` capture function name, so my linter was complaining; figured it was worth fixing

**Review by rntz — APPROVED** [review]
Looks good, I haven't tested it though.

**Comment by pokey** [comment]
Works fine for me; I don't think it's going to vary by platform or anything so probably good to just merge?

File: core/numbers/numbers.py (hunk 1 above)
Comment (AndreasArvidsson) [reply core/numbers/numbers.py:8]: Must say I'm not a big fan of captures with plural names that only return a single value. I call my alternative just `digit`

File: core/numbers/numbers.py [hunk 2]
```diff
@@ -5,12 +5,12 @@
...
 teens = "ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen".split()
 tens = "twenty thirty forty fifty sixty seventy eighty ninety".split()
 scales = "hundred thousand million billion trillion quadrillion quintillion sextillion septillion octillion nonillion decillion".split()
 
-digits_map = {n: i for i, n in enumerate(digits)}
+digits_map = {n: i for i, n in enumerate(digit_list)}
 digits_map["oh"] = 0
```
Comment (rntz) [inline core/numbers/numbers.py:14]: Shit, I didn't realize it, but this is a problem. Your new version uses `digits_list` rather than `digits_map` and therefore doesn't include any small numbers that use "oh" for zero! So merging this PR broke backwards compat, my bad.

The data flow through globals in this file is getting kinda hard to follow, we should probably try to confine it to just one place.
